packages/react-dom/src/client/ReactInputSelection.js

```diff
@@ -116,7 +116,13 @@ export function restoreSelection(priorSelectionInformation) {
   const curFocusedElem = getActiveElementDeep();
   const priorFocusedElem = priorSelectionInformation.focusedElem;
   const priorSelectionRange = priorSelectionInformation.selectionRange;
-  if (curFocusedElem !== priorFocusedElem && isInDocument(priorFocusedElem)) {
+  // With disabled elements, they are not focusable
+  if (
+    priorFocusedElem &&
+    !priorFocusedElem.disabled &&
```

Comment:
Shouldn't this check instead be just before the `focus` call so that the rest of the logic runs? Such as restoring the scroll positions.

Reply:
I've been thinking about it for quite some time but it may seems a bit useless, let me explain.

The fact is a disabled element can not be focused, focusing a disabled element will always focus the body (I'm checking it by `document.activeElement`). Since it returns a wrong result, the logic below might as well be not properly applied.

Now let's take a look at the logic of the function. It's restoring selection. However, the selection of a disabled element can't be restored too. Because it requires a focus (and this focus point to body instead). I'm testing with an `input` with `setSelectionRange` function.

The funniest thing here is that it's not the focus function call blocks the scroll. Indeed, it's the restoring of the scroll position cause the bug (the logic inside the for loop just below the focus calling). I guess because there is no position should be restored in this case (If I understand correctly, this restore should be applied only after mutation phase that involves deletion). Making a check before the focus won't solve the problem. Instead, we can make a check just before the calling of a for loop and the problem should be resolved but the above code will be kinda useless. 

After all, with careful consideration, I decided to put the conditional checking to the first `if` which is in this PR.

Thanks for your review :)